### PR TITLE
[WIP] Component tests for BWFilter and SepiaFilter.

### PR DESCRIPTION
**Pull request by sminnee**
This adds component tests for BWFilter and SepiaFilter, for both the GD
and Imagick	backends. These tests execute the conversion and compare the
results to reference images, rather than just mocking out the relevant
backend function calls.

Writing these tests exposed an additional weakness, namely that GD and
Imagick output quite different images. For now, I have made this
apparent by introducing a pair of reference images. It would be nice to
alter the implementation of one of the backends so that the output was
more consistent across both.

### CODE REVIEW

**Comment by tiemevanveen** [comment]
Nice! I'll add these for the Gaussian filter as wel!

**Comment by erikaheidi** [comment]
We still need to work a bit more on this, it would be nice to make sure the filter gets a specific result but I think it won't be a easily scalable solution unfortunately... as we saw, different versions of Imagick will give different image results.

**Comment by sminnee** [comment]
@erikaheidi sure thing. could you give me details of the sandbox environment; I never got that set up at the CodeNight.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@ build
 vendor
 examples
 composer.lock
+.DS_Store
```
Comment (ProPheT777) [inline .gitignore]: :scream_cat:

**Pull request closed without merging**